src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensEditsHandler.cs

```diff
@@ -44,9 +44,15 @@ public SemanticTokensEditsHandler(SemanticTokensCache tokensCache)
             RequestContext context,
             CancellationToken cancellationToken)
         {
+            // Temporary workaround for https://github.com/dotnet/roslyn/issues/54547:
+            // We should eventually go back to throwing here if context.Document is null.
+            if (context.Document is null)
```

Comment:
oh... boy do i hate this.  IMO this is not ok and we need to revert this PR.  This is precisely teh sort of "limp along when we dont' know what is happening" approach that leads to extreme amounts of tech debt and an inability to reason about what's going on.